src/events.rs

```diff
@@ -210,10 +210,11 @@ impl EventAssert {
 
 /// supervisor_started asserts an event that tells a supervisor with the given
 /// name started
-pub fn supervisor_started(input_name: &'static str) -> EventAssert {
+pub fn supervisor_started(input_name0: &str) -> EventAssert {
+    let input_name = input_name0.to_owned();
     EventAssert(Box::new(move |ev| match &ev {
         Event::SupervisorStarted(NodeData { runtime_name }) => {
-            if runtime_name != input_name {
+            if runtime_name != &*input_name {
```

Comment:
This is a bit curious to me. Aren't both `runtime_name` and `input_name` `String`s? I guess in this context `runtime_name` is borrowed because you match on `&ev`, but I think you should be able to just use the `&` without the `*`. Maybe I'm wrong tho :)

Reply:
That was the only way I could make it compile

![no idea](https://user-images.githubusercontent.com/7335/93402979-19eddf80-f83b-11ea-9056-9ef84c2d5709.png)

Reply:
Compilation error I get if I don't do `&*`

```
   Compiling capataz v0.0.0 (/home/roman/tmp/rust-capataz)
error[E0277]: can't compare `&std::string::String` with `std::string::String`
   --> src/events.rs:311:29
    |
311 |             if runtime_name != input_name {
    |                             ^^ no implementation for `&std::string::String == std::string::String`
    |
    = help: the trait `std::cmp::PartialEq<std::string::String>` is not implemented for `&std::string::String`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `capataz`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error[E0277]: can't compare `&std::string::String` with `std::string::String`
   --> src/events.rs:311:29
    |
311 |             if runtime_name != input_name {
    |                             ^^ no implementation for `&std::string::String == std::string::String`
    |
    = help: the trait `std::cmp::PartialEq<std::string::String>` is not implemented for `&std::string::String`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: build failed
```

Reply:
Ohhhh, right. there's a closure involved my bad. You're doing it right :sweat_smile:

For the comparison though you should just deref (`*`) `runtime_name, that should make the comparison work.

Reply:
Actually, that's not true, the deref won't work probably. I think the `&*` is fine and probably really the only way to make it work...